core-tests/src/test/java/org/mule/runtime/core/DefaultMessageContextTestCase.java

```diff
@@ -97,4 +98,23 @@ public void overrideCorrelationIdInContextSequence() {
     assertThat(event.getCorrelationId(), is(CUSTOM_CORRELATION_ID));
   }
 
+  @Test
+  public void overrideCorrelationIdInContext() {
+    final Message message = of(TEST_PAYLOAD);
+    final CoreEvent event = CoreEvent.builder(executionContextWithCorrelation).message(message)
+        .itemSequenceInfo(empty()).build();
+
+    assertThat(event.getCorrelationId(), is(CUSTOM_CORRELATION_ID));
+  }
+
+  @Test
+  public void overrideCorrelationIdInContextSequence() {
+    final Message message = of(TEST_PAYLOAD);
+    final CoreEvent event =
+        PrivilegedEvent.builder(executionContextWithCorrelation).message(message).correlationId(CUSTOM_CORRELATION_ID)
+            .itemSequenceInfo(Optional.of(ItemSequenceInfo.of(6))).build();
+
+    assertThat(event.getCorrelationId(), is(CUSTOM_CORRELATION_ID));
+  }
+
```

Comment:
I don't quite understand these tests.
Are their names accurate? The first one doesn't really seem to be overriding anything.
Also, shouldn't they be asserting something related to the `itemSequenceInfo` besides the `correlationId`?

Reply:
this are just the same tests as above but using itemSequenceInfo instead of groupCorrelation